rust/trace/src/chrome_trace_dump.rs

```diff
@@ -82,6 +83,36 @@ where
     serde_json::from_reader(input).map_err(Error::Json)
 }
 
+/// Save tracing data to to supplied path, using the Trace Viewer format. Trace file can be opened
+/// using the Chrome browser by visiting the URL `about:tracing`. If `sorted_chronologically` is
+/// true then sort output traces chronologically by each trace's time of creation.
+pub fn save_traces(output_path: &Path, sorted_chronologically: bool) -> IOResult<()> {
+    let traces = if sorted_chronologically {
+        super::samples_cloned_sorted()
+    } else {
+        super::samples_cloned_unsorted()
+    };
+
+    match File::create(&output_path) {
```

Comment:
I would return the error directly (`File::create(&output_path)?`) and let the caller handle it.

Reply:
Additionally, we should at least think about what the correct behaviour is in the case where the path exists.

Reply:
@cmyr I gave some thought to what correct behaviour should be in the case where the path (file) already exists:

- we could overwrite the file: this is dangerous, since an important file could be overwritten.
- we could rename the file name component of the path: in the end, in order to guarantee that we come up with a unique file name, we cannot rely on any methods that use randomness at some level to generate a file name (as there would always exist some possibility of failure). We must instead list every file name in the directory containing the file, come up with a file name (possibly through some random means), and ensure it is not in the already tabulated list of existing file.
- we could do nothing, and abort (return an error): this is robust, in the sense that `save_traces` will definitely not screw up in any important way, and it is up to the caller to ensure they are providing okay file names.

I have chosen to go with the last option, for now. Let me know what you think.